### PR TITLE
feat(terraform): add CKV NCP rule about block storage encryption.

### PR DESCRIPTION
**Pull request by Floodnut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Add terraform scan rule about provider Naver Cloud Platform.

CKV_NCP_6 is associated with NCloud Server Encryption.
This rule suggests that base block storage should be encrypted.

### Description
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/server

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class ServerEncryptionVPC(BaseResourceCheck):
+    def __init__(self):
+        name = "You can set whether to encrypt basic block storage if server image is RHV. Default false." \
+               " Only support VPC environment."
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py:9]:
```suggestion
        name = "Ensure Server instance is encrypted"
```

File: checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class ServerEncryptionVPC(BaseResourceCheck):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py:6]:
```suggestion
from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck


class ServerEncryptionVPC(BaseResourceValueCheck):
```

File: checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    def scan_resource_conf(self, conf):
+        if 'is_encrypted_base_block_storage_volume' in conf.keys():
+            is_encrypted_base_block_storage_volume = conf['is_encrypted_base_block_storage_volume']
+            if is_encrypted_base_block_storage_volume == [True]:
+                return CheckResult.PASSED
+            else:
+                return CheckResult.FAILED
+        else:
+            return CheckResult.FAILED
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/ServerEncryptionVPC.py:23]:
```suggestion
    def get_inspected_key(self):
        return "is_encrypted_base_block_storage_volume"
```

**Review by gruebel — CHANGES_REQUESTED** [review]
such a check can be easily written by subclassing `BaseResourceValueCheck` instead of `BaseResourceCheck`

**Comment by Floodnut** [comment]
I adjust it!

**Comment by gruebel** [comment]
please merge master again into your branch, I tried to do it, but got a permission denied

**Comment by Floodnut** [comment]
> please merge master again into your branch, I tried to do it, but got a permission denied

Oh, sorry. merge this branch again!

**Review by gruebel — APPROVED** [review]
thanks again 🍺

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**